src/server/command/worldedit/blueprint.zig

```diff
@@ -0,0 +1,198 @@
+		.list => blueprintList(source),
+		.unknown => {
+			source.sendMessage("#ff0000Unrecognized subcommand for /blueprint: '{s}'", .{argsList.items[0]});
+		},
+		.empty => {
+			source.sendMessage("#ff0000Missing subcommand for /blueprint, usage: {s} ", .{usage});
+		},
+	}
+}
+
+fn blueprintSave(args: []const []const u8, source: *User) void {
+	if(args.len < 2) {
+		source.sendMessage("#ff0000/blueprint save requires file-name argument.", .{});
+		return;
+	}
+	if(args.len >= 3) {
+		source.sendMessage("#ff0000Too many arguments for /blueprint save. Expected 1 argument, file-name.", .{});
+		return;
+	}
+
+	if(source.worldEditData.clipboard) |clipboard| {
+		const storedBlueprint = clipboard.store(main.stackAllocator);
+		defer main.stackAllocator.free(storedBlueprint);
+
+		const fileName: []const u8 = ensureBlueprintExtension(main.stackAllocator, args[1]);
+		defer main.stackAllocator.free(fileName);
+
+		var blueprintsDir = openBlueprintsDir(source) orelse return;
+		defer blueprintsDir.close();
+
+		blueprintsDir.write(fileName, storedBlueprint) catch |err| {
+			sendWarningAndLog("Failed to write blueprint file '{s}' ({s})", .{fileName, @errorName(err)}, source);
+			return;
+		};
+
+		sendInfoAndLog("Saved clipboard to blueprint file: {s}", .{fileName}, source);
+	} else {
+		source.sendMessage("#ff0000Error: No clipboard content to save.", .{});
+	}
+}
+
+fn sendWarningAndLog(comptime fmt: []const u8, args: anytype, user: *User) void {
+	std.log.warn(fmt, args);
+	user.sendMessage("#ff0000" ++ fmt, args);
+}
+
+fn sendInfoAndLog(comptime fmt: []const u8, args: anytype, user: *User) void {
+	std.log.warn(fmt, args);
+	user.sendMessage("#00ff00" ++ fmt, args);
+}
+
+fn openBlueprintsDir(source: *User) ?Dir {
+	return openDir("blueprints") catch |err| blk: {
+		sendWarningAndLog("Failed to open 'blueprints' directory ({s})", .{@errorName(err)}, source);
+		break :blk null;
+	};
+}
+
+fn ensureBlueprintExtension(allocator: NeverFailingAllocator, fileName: []const u8) []const u8 {
+	if(!std.ascii.endsWithIgnoreCase(fileName, ".blp")) {
+		return std.fmt.allocPrint(allocator.allocator, "{s}.blp", .{fileName}) catch unreachable;
+	} else {
+		return allocator.dupe(u8, fileName);
+	}
+}
+
+fn blueprintDelete(args: []const []const u8, source: *User) void {
+	if(args.len < 2) {
+		source.sendMessage("#ff0000/blueprint delete requires file-name argument.", .{});
+		return;
+	}
+	if(args.len >= 3) {
+		source.sendMessage("#ff0000Too many arguments for /blueprint delete. Expected 1 argument, file-name.", .{});
+		return;
+	}
+
+	const fileName: []const u8 = ensureBlueprintExtension(main.stackAllocator, args[1]);
+	defer main.stackAllocator.free(fileName);
+
+	var blueprintsDir = openBlueprintsDir(source) orelse return;
+	defer blueprintsDir.close();
+
+	blueprintsDir.dir.deleteFile(fileName) catch |err| {
+		sendWarningAndLog("Failed to delete blueprint file '{s}' ({s})", .{fileName, @errorName(err)}, source);
+		return;
```

Comment:
Not sure if you want to do this, but you could also put these in one line, since the return types (void) match:
```
return sendWarningAndLog(...);
```